### PR TITLE
fix(core): socket pushes not working

### PR DESCRIPTION
**Pull request by kkopanidis**
The io object seems weird in its behaviour,
but the changes should allow for connected
sockets to receive the data correctly

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### CODE REVIEW

**Comment by kkopanidis** [comment]
Proper testing is needed, I'm still facing network issues.

File: modules/chat/src/routes/routes.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ export class ChatRoutes {
 
   async onMessage(call: ParsedSocketRequest): Promise<UnparsedSocketResponse> {
     const { user } = call.request.context;
-    const [roomId, message] = call.request.params;
+    const { roomId, message } = call.request.params;
```
Comment (kon14) [inline modules/chat/src/routes/routes.ts:302]: This should be `[ roomId, message ]`. Socket params are an array.

File: modules/chat/src/routes/routes.ts [hunk 2]
```diff
@@ -323,7 +323,7 @@ export class ChatRoutes {
 
   async onMessagesRead(call: ParsedSocketRequest): Promise<UnparsedSocketResponse> {
     const { user } = call.request.context;
-    const [roomId] = call.request.params;
+    const { roomId } = call.request.params;
```
Comment (kon14) [inline modules/chat/src/routes/routes.ts:326]: Same ^

**Review by kon14 — CHANGES_REQUESTED** [review]
ParsedSocketRequest's `call.request.params` are an array.
Object destructuring will produce `undefined`.

**Comment by kkopanidis** [comment]
@kon14 slow down, I'm looking at these right now, I only resolved conflicts previously. Even without the resolution it wouldn't work there are multiple typing issues.

**Review by kon14 — APPROVED** [review]

**Pull request merged**